src/fides/core/dataset.py

```diff
@@ -373,3 +377,33 @@ def generate_dynamo_db_datasets(aws_config: Optional[AWSConfig]) -> Dataset:
     described_dynamo_tables = describe_dynamo_tables(client, dynamo_tables)
     dynamo_dataset = create_dynamodb_dataset(described_dynamo_tables)
     return dynamo_dataset
+
+
+def get_snowflake_schemas(
+    engine: Engine,
+) -> Dict[str, Dict[str, List[str]]]:
+    """
+    Returns Datasets that match the case-sensitivity that may be
+    required by Snowflake.
```

Comment:
nit - may be helpful to provide a bit more explanation here? specifically - what about this differs from the "standard"  `inspector.get_schema_names()`, etc. calls being made on the other codepath? or is there any documentation/reference that explains why we need to do this? 

granted i'm looking at this code with not a lot of context/experience with `sqlalchemy.inspect` so maybe that's why it's just a bit hard for me to understand what exactly we're doing differently here to ensure we get case-sensitive metadata. i'm just imagining coming back to this code in some months/years and looking for a bit more context 😄

Reply:
No this makes total sense! A common first impression of people using Snowflake is some general surprise to their defaulting of casing to all caps. What makes this extra interesting is the officially supported sqlalchemy dialect then has the `inspect()` method default everything to lower case and the issues to address this are more than three years stale at this point. 

I'll add this context in the docs block of the code so that we can have it closer to the issue, let me know if that helps cover that a bit more and thanks for the suggestion 👍🏽

Reply:
yup that sounds great!